File: subtract/func.rb
require 'fdk'
require 'bigdecimal'

def subtract(context:, input:)
  left = input.respond_to?(:fetch) ? input.fetch('left') : "0"
  right = input.respond_to?(:fetch) ? input.fetch('right') : "0"
  l = BigDecimal(left)
  r = BigDecimal(right)
  { result: (l - r).to_s("F") }
end

FDK.handle(target: :subtract)

File: plus/func.js
const fdk = require('@fnproject/fdk');

fdk.handle(function (input) {
  let l = 0
  let r = 0
  if (input.left) {
    l = Number(input.left)
  }
  if (input.right) {
    r = Number(input.right)
  }
  const result = l + r
  return { 'result': String(result) }
})
